devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,680 @@
+		def backlightNetworkId = "${device.deviceNetworkId}:${i}"
+		def childRelayExists = (existingChildren.find {child -> child.getDeviceNetworkId() == relayNetworkId} != NULL)
+		def childButtonExists = (existingChildren.find {child -> child.getDeviceNetworkId() == buttonNetworkId} != NULL)
+		def childBacklightExists = (existingChildren.find {child -> child.getDeviceNetworkId() == backlightNetworkId} != NULL)
+		if (!childBacklightExists ) {
+			def child = addChildDevice("Heltun Child Backlight", backlightNetworkId, device.hubId, [completedSetup: true, label: getChildName(i), isComponent: false])
+		}
+		if (!childRelayExists) {
+			addChildDevice("HELTUN", "Heltun Child Relay", relayNetworkId, device.hubId,[completedSetup: true, label: getChildName(i+numberOfButtons), isComponent: false])
+		}
+		if (!childButtonExists ) {
+			def child = addChildDevice("smartthings", "Child Button", buttonNetworkId, device.hubId, [completedSetup: true, label: getChildName(i+2*numberOfButtons), isComponent: true, componentName: "button$i", componentLabel: "Button ${i}"])
+		}
+	}
+	initialize()
+}
+
+private getChildName(channelNumber) {
+	def numberOfButtons = state.numberOfButtons
+	if (channelNumber in 1..numberOfButtons) {
+		return "${device.displayName} " + "${"Backlight"} " + "${channelNumber}"
+	}
+	else if (channelNumber in (numberOfButtons+1)..(2*numberOfButtons)){
+		return "${device.displayName} " + "${"Switch"} " + "${channelNumber-numberOfButtons}"
+	}
+	else if (channelNumber in (2*numberOfButtons+1)..(3*numberOfButtons)){
+		return "${device.displayName} " + "${"Button"} " + "${channelNumber-numberOfButtons*2}"
+	}
+}
+
+private channelNumber(String deviceNetworkId) {
+	deviceNetworkId.split(":")[-1] as Integer
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		return zwaveEvent(cmd)
+	}
+}
+
+private void setState(value, endpoint = null) {
+	def map = [
+		encap(zwave.basicV1.basicSet(value: value), endpoint),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), endpoint),
+	]
+	sendHubCommand(map, 500)
+}
+
+private encap(cmd, endpoint) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+def childOn(childId) {
+	setState(0xFF, channelNumber(childId))
+}
+
+def childOff(childId) {
+	setState(0, channelNumber(childId))
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def map = [:]
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			map.name = "energy"
+			map.value = cmd.scaledMeterValue
+			map.unit = "kWh"
+			sendEvent(map)
+		} else if (cmd.scale == 2) {
+			map.name = "power"
+			map.value = Math.round(cmd.scaledMeterValue)
+			map.unit = "W"
+			sendEvent(map)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	def numberOfButtons = state.numberOfButtons
+	def state
+	def buttonN
```

Comment:
Can't you just do `def buttonN = cmd.sceneNumber` here? Or you're expecting not wanted `cmd.keyAttributes` which you'd prefer to ignore?